hammers/scripts/lease_stack_reaper.py

```diff
@@ -206,43 +193,38 @@ def main(argv=None):
     args = parser.parse_args(argv[1:])
     mysqlargs.extract(args)
     auth = osapi.Auth.from_env_or_args(args=args)
-
-    if args.slack:
-        slack = Slackbot(args.slack, script_name='lease_stack_reaper')
-    else:
-        slack = None
+    sender = args.sender
+    slack = Slackbot(args.slack, script_name='lease_stack_reaper') if args.slack else None
 
     db = mysqlargs.connect()
     db.version = query.ROCKY
 
-    kwargs = {
-        'db': db,
-        'auth': auth,
-        'describe': args.action == 'info',
-        'quiet': args.quiet,
-        'sender': args.sender
-    }
-
-    if slack:
-        with slack:
-            remove_count = lease_stack_reaper(**kwargs)
-    else:
-        remove_count = lease_stack_reaper(**kwargs)
-
-    if slack and (args.action == 'delete') and (
-            (not args.quiet) or remove_count):
-
-        if remove_count > 0:
-            message = (
-                'Commanded deletion of *{} leases* '
-                '(Lease stacking restriction violated)'
-                .format(remove_count)
-            )
-            color = '#000000'
-        else:
-            message = ('No leases to delete.')
-            color = '#cccccc'
-        slack.post('lease-stack-reaper', message, color=color)
+    try:
+        users = collect_user_leases(db)
```

Comment:
Changed this to return the user/lease structures, then we decide what to do based on `args.action`.